### PR TITLE
[Merged by Bors] - refactor(LinearAlgebra/BilinearForm/Properties): Redefine BilinForm.IsRefl, BilinForm.IsAlt and BilinForm.IsSymm in terms of LinearMap equivalents 

### PR DESCRIPTION
**Pull request by mans0954**
Redefine `BilinForm.IsRefl`, `BilinForm.IsAlt` and `BilinForm.IsSymm` in terms of `LinearMap.IsRefl` and `LinearMap.IsSymm`.

---

- [x] depends on: #10422

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/BilinearForm/Properties.lean [hunk 1]
```diff
@@ -55,8 +54,7 @@ namespace BilinForm
 
 
 /-- The proposition that a bilinear form is reflexive -/
-def IsRefl (B : BilinForm R M) : Prop :=
-  ∀ x y : M, B x y = 0 → B y x = 0
+def IsRefl (B : BilinForm R M) : Prop := LinearMap.IsRefl (toLin B)
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/BilinearForm/Properties.lean:57]: Can the `toLin`s be dropped now that the other PR landed?

Comment (mans0954) [reply Mathlib/LinearAlgebra/BilinearForm/Properties.lean]: Yes. In fact `toLin` is deprecated in #12078.

File: Mathlib/LinearAlgebra/BilinearForm/Orthogonal.lean [hunk 2]
```diff
@@ -166,7 +166,7 @@ theorem mem_orthogonal_iff {N : Submodule R M} {m : M} :
 theorem orthogonal_le (h : N ≤ L) : B.orthogonal L ≤ B.orthogonal N := fun _ hn l hl => hn l (h hl)
 #align bilin_form.orthogonal_le LinearMap.BilinForm.orthogonal_le
 
-theorem le_orthogonal_orthogonal (b : B.IsRefl) : N ≤ B.orthogonal (B.orthogonal N) :=
+theorem le_orthogonal_orthogonal (b : B.IsRefl) : N₂ ≤ B.orthogonal (B.orthogonal N₂) :=
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/BilinearForm/Orthogonal.lean:169]: Why the change to N2 here?

Comment (mans0954) [reply Mathlib/LinearAlgebra/BilinearForm/Orthogonal.lean]: Good question! I think it's a hangover from before we made `R` commutative https://github.com/leanprover-community/mathlib4/commit/347b765edc635a9b8fa132661b2dc3e2c3e29fc1 I've removed it in favour of `N` now.

**Comment by eric-wieser** [comment]
bors d+

Sorry for forgetting about this one 

Is the description still correct? You already did replace BilinForm, right?

**Comment by mans0954** [comment]
> Is the description still correct? You already did replace BilinForm, right?

I opened this PR before we did that! I think the PR is still useful for reducing duplication though. I've updated the description. Thanks.
